model/lifecycle.go

```diff
@@ -1297,6 +1267,18 @@ func createOneTask(id string, buildVarTask BuildVariantTaskUnit, project *Projec
 		// TODO (EVG-16169): set ExecutionPlatform based on whether or not the task
 		// is running in a host or container.
 		ExecutionPlatform: task.ExecutionPlatformHost,
+		RunOnContainer:    utility.FromBoolPtr(buildVarTask.RunOnContainer),
```

Comment:
It seems like this field is redundant since the `ExecutionPlatform` field above this has the same meaning. I would just set `ExecutionPlatform` based on whether RunOn is for a container or distros (which also fulfills EVG-16169).